tests/include/mir/test/barrier.h

```diff
@@ -38,7 +38,7 @@ class Barrier
     void ready()
     {
         std::unique_lock lock(mutex);
-        if (--wait_threads)
+        if ((--wait_threads) > 0)
```

Comment:
You realise that's an implicit conversion of 0 to unsigned? That just makes the code confusing

Reply:
Oh whoops you are write!